src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java

```diff
@@ -162,7 +162,7 @@ private static boolean startsWithInheritDoc(DetailNode root) {
         boolean found = false;
         final DetailNode[] children = root.getChildren();
 
-        for (int i = 0; !found && i < children.length - 1; i++) {
+        for (int i = 0; !found; i++) {
```

Comment:
We always break before the end, so this check was removed. If this was not the case, we would eventually get an out of index exception.